toolchain/check/testdata/function/declaration/fail_modifiers.carbon

```diff
@@ -70,7 +70,7 @@ base fn InvalidModifier();
 // CHECK:STDERR: ^~~~~~~
 default final virtual fn ModifiersConflict2() {}
 
-// CHECK:STDERR: fail_modifiers.carbon:[[@LINE+9]]:1: ERROR: `extern` not allowed on `fn` declaration.
+// CHECK:STDERR: fail_modifiers.carbon:[[@LINE+9]]:1: ERROR: `extern` not allowed on `fn` declaration on a definition.
```

Comment:
Similar but a bit worse I think here is the context string not working well. Keeping discussion focused on the above comment.